examples/userapi/05_conditions.ipynb

```diff
@@ -0,0 +1,489 @@
+      "} ;\n",
+      "\n",
+      "\n",
+      "int Kernel(struct dataobj *restrict f_vec, const int i_M, const int i_m, const int j_M, const int j_m, struct profiler * timers, const int nthreads)\n",
+      "{\n",
+      "  float (*restrict f)[f_vec->size[1]] __attribute__ ((aligned (64))) = (float (*)[f_vec->size[1]]) f_vec->data;\n",
+      "  /* Flush denormal numbers to zero in hardware */\n",
+      "  _MM_SET_DENORMALS_ZERO_MODE(_MM_DENORMALS_ZERO_ON);\n",
+      "  _MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);\n",
+      "  struct timeval start_section0, end_section0;\n",
+      "  gettimeofday(&start_section0, NULL);\n",
+      "  /* Begin section0 */\n",
+      "  #pragma omp parallel num_threads(nthreads)\n",
+      "  {\n",
+      "    #pragma omp for collapse(1) schedule(static,1)\n",
+      "    for (int i = i_m; i <= i_M; i += 1)\n",
+      "    {\n",
+      "      #pragma omp simd aligned(f:32)\n",
+      "      for (int j = j_m; j <= j_M; j += 1)\n",
+      "      {\n",
+      "        if (f[i][j] > 0 && f[i][j] < 5)\n",
+      "        {\n",
+      "          f[i][j] = f[i][j] + 1;\n",
+      "        }\n",
+      "      }\n",
+      "    }\n",
+      "  }\n",
+      "  /* End section0 */\n",
+      "  gettimeofday(&end_section0, NULL);\n",
+      "  timers->section0 += (double)(end_section0.tv_sec-start_section0.tv_sec)+(double)(end_section0.tv_usec-start_section0.tv_usec)/1000000;\n",
+      "  return 0;\n",
+      "}\n",
+      "\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "Data([[3., 2., 2., 2., 2., 2., 2., 2., 2., 2.],\n",
+       "      [2., 3., 2., 2., 2., 2., 2., 2., 2., 2.],\n",
+       "      [2., 2., 3., 2., 2., 2., 2., 2., 2., 2.],\n",
+       "      [2., 2., 2., 3., 2., 2., 2., 2., 2., 2.],\n",
+       "      [2., 2., 2., 2., 3., 2., 2., 2., 2., 2.],\n",
+       "      [2., 2., 2., 2., 2., 3., 2., 2., 2., 2.],\n",
+       "      [2., 2., 2., 2., 2., 2., 3., 2., 2., 2.],\n",
+       "      [2., 2., 2., 2., 2., 2., 2., 3., 2., 2.],\n",
+       "      [2., 2., 2., 2., 2., 2., 2., 2., 3., 2.],\n",
+       "      [2., 2., 2., 2., 2., 2., 2., 2., 2., 3.]], dtype=float32)"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from devito import ConditionalDimension\n",
+    "from devito.symbolics import CondEq\n",
+    "from sympy import And\n",
+    "\n",
+    "condition = And(f[i,j] > 0, f[i,j] < 5 , evaluate=False)\n",
+    "ci = ConditionalDimension(name='ci', parent=j, condition=condition)\n",
+    "\n",
+    "op = Operator(Eq(f, f + 1, implicit_dims=ci))\n",
+    "op.apply()\n",
+    "print(op.ccode)\n",
+    "f.data[:, :]\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Our operator will be increasing by one as far as element value does not get larger than 5. You can execute the following cell a few times to notice this. All values will converge to 5."
```

Comment:
Wording, e.g. 'The affect of this operator will be to increase values of our function `f` by 1 provided that the current value of `f(i,j)` is less than 5. Hence, following (and beyond) a sufficient number of applications of the operator we expect all values of `f` to be come 5.'

Then actually show the converged state - no need to show any more intermediate states.